app/controllers/home_controller.rb

```diff
@@ -73,12 +72,18 @@ def search_params
 
   def search_and_return
     if @search_form.valid?
-      begin
-        OnlineApplication.find_by!(reference: @search_form.reference.upcase)
-      rescue ActiveRecord::RecordNotFound
-        @search_form.errors.add(:reference, :not_found)
-        nil
+      @search = ApplicationSearch.new(@search_form.reference.upcase, current_user)
```

Comment:
The `@search` here could be just a local variable, it's not exposed to the view.

Also I think it's cleaner if you `upcase` the reference inside the service, it keeps the requirements on the interface cleaner.